### PR TITLE
display call stack for unsuccessful unsafe calls

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

# Overview

- [x] I have added "stack traces" for unsafe calls when they fail.

# Comments

It's not much, but it turns:

```
       uncaught exception: ErrorCall
       TextDecodingError {getTextDecodingError = "Unable to decode Address: encoding is neither Bech32 nor Base58."}
       CallStack (from HasCallStack):
         error, called at src/Cardano/Wallet/Unsafe.hs:56:13 in cardano-wallet-core-2019.7.24-vtyOM1MPP4BB3M0K9gXCg:Cardano.Wallet.Unsafe
```

into

```
       uncaught exception: ErrorCall
       TextDecodingError {getTextDecodingError = "Unable to decode Address: encoding is neither Bech32 nor Base58."}
       CallStack (from HasCallStack):
         error, called at src/Cardano/Wallet/Unsafe.hs:58:13 in cardano-wallet-core-2019.7.24-vtyOM1MPP4BB3M0K9gXCg:Cardano.Wallet.Unsafe
         unsafeDecodeAddress, called at test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs:121:21 in main:Cardano.Wallet.Jormungandr.CompatibilitySpec
```

which makes it easier to locate the point of failure.

### CODE REVIEW

**Review by Anviking — APPROVED** [review]
👍

**Review by paweljakubas — APPROVED** [review]

**Comment by KtorZ** [comment]
bors r_

**Review by rvl — APPROVED** [review]
Good idea :+1:.
Too bad we can't use it on `invariant` without applying `HasCallStack` to basically the whole app.

File: lib/core/src/Cardano/Wallet/Unsafe.hs [hunk 1]
```diff
@@ -46,29 +49,30 @@ import qualified Data.ByteString.Lazy as BL
...
 
 -- | Decode an hex-encoded 'ByteString' into raw bytes, or fail.
-unsafeFromHex :: ByteString -> ByteString
+unsafeFromHex :: HasCallStack => ByteString -> ByteString
 unsafeFromHex =
     either (error . show) id . convertFromBase @ByteString @ByteString Base16
 
 -- | Decode a bech32-encoded 'Text' into an 'Address', or fail.
-unsafeDecodeAddress :: DecodeAddress t => Proxy t -> Text -> Address
+unsafeDecodeAddress :: (HasCallStack, DecodeAddress t) => Proxy t -> Text -> Address
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet/Unsafe.hs:57]:
```suggestion
unsafeDecodeAddress
    :: (HasCallStack, DecodeAddress t) => Proxy t -> Text -> Address
```

**Review by jonathanknowles — APPROVED** [review]
Looks like a nice idea, when used sparingly in places where it will help.

Only one minor change suggested.

**Pull request merged**